password_security/models/res_users.py

```diff
@@ -81,6 +81,8 @@ def check_password(self, password):
             password_regex.append(r'(?=.*?\W)')
         password_regex.append('.{%d,}$' % company_id.password_length)
         if not re.search(''.join(password_regex), password):
+            self.env.cr.rollback()
```

Comment:
I feel all is much more clean with `with self.env.cr.savepoint():`, which rolls back automatically if any exception is found inside the block (the same for all cases below).